lms/djangoapps/verify_student/views.py

```diff
@@ -1175,15 +1179,69 @@ def post(self, request, course_id, checkpoint_name):
                 EVENT_NAME_USER_SUBMITTED_INCOURSE_REVERIFY, user.id, course_id, checkpoint_name
             )
 
-            return HttpResponse()
+            redirect_url = self._get_redirect_url(course_id, location)
+            return JsonResponse({
+                'url': redirect_url
+            })
+        except Http404 as e:
+            log.exception("Invalid location during photo verification.")
+            return HttpResponseBadRequest(_(e.message))
         except IndexError:
             log.exception("Invalid image data during photo verification.")
             return HttpResponseBadRequest(_("Invalid image data during photo verification."))
         except Exception:  # pylint: disable=broad-except
-            log.exception("Could not submit verification attempt for user {}.").format(request.user.id)
+            log.exception("Could not submit verification attempt for user {}.".format(request.user.id))
             msg = _("Could not submit photos")
             return HttpResponseBadRequest(msg)
 
+    def _get_redirect_url(self, course_id, location):
+        """ Returns the redirect url back to courseware
+
+        Args:
+            course_id(str): Course Id string
+            location(str): The location id of course component
+
+        Raises:
+            Https404 if invalid course_id or invalid location
+
+        Returns:
+            Redirect url string
+        """
+        try:
+            course_key = CourseKey.from_string(course_id)
+            usage_key = UsageKey.from_string(location).replace(course_key=course_key)
+        except InvalidKeyError:
+            raise Http404(u"Invalid course_key or usage_key")
+        try:
+            (course_key, chapter, section, position) = path_to_location(modulestore(), usage_key)
+        except ItemNotFoundError:
+            raise Http404(u"No data at this location: {0}".format(usage_key))
+        except NoPathToItem:
+            raise Http404(u"This location is not in any class: {0}".format(usage_key))
```

Comment:
unfortunately there is a wider pattern in the lms of raising HTTP errors in functions that are not themselves views.  if we want to reuse this function in courseware and preserve existing courseware behavior it will be necessary to extract the exception handling / HTTP errors out into whatever views are already calling the function.